src/main/java/Sprites.java

```diff
@@ -0,0 +1,112 @@
+import org.json.JSONObject;
+import org.json.JSONArray;
+import java.util.Map;
+import java.util.HashMap;
+
+/**
+ * @version 1.0
+ * @author B. Clint Hall
+ */
+public class Sprites {
+    private Map<String, JSONObject> spriteMap;
+    /**
+     * The root of the sb2 json object is a Stage object.
+     * This constructs a Sprites object from a Stage JSONObject.
+     * @param stage root of an Sb2 file
+     */
+    public Sprites(JSONObject stage) {
+        JSONArray stageChildren = stage.optJSONArray("children");
+        if (stageChildren == null) {
+            stageChildren = new JSONArray();
+        }
+        spriteMap = new HashMap<String, JSONObject>();
+        for (int i = 0; i < stageChildren.length(); i++) {
+            JSONObject child = stageChildren.optJSONObject(i);
+            if (isSprite(child)) {
+                spriteMap.put(child.getString("objName"), child);
+            }
+        }
+    }
+    /**
+     * Check whether a stage child is a Sprite.  It could also be
+     * a StageMonitor.
+     * @param stageChild A JSONObject that may represent a Sprite or a
+     *        StageMonitor
+     * @return whether the stageChild is a Sprite.
+     */
+    private boolean isSprite(JSONObject stageChild) {
+        return stageChild != null && stageChild.has("spriteInfo");
+    }
+    /**
+     * Each sprite has a unique name.
+     * This method returns these names in a array of strings.
+     * @return Array of sprite names.
+     */
+    public String[] getSpriteNames() {
+        String[] spriteNames = new String[spriteMap.size()];
+        return spriteMap.keySet().toArray(spriteNames);
+    }
+
+
+    /**
+     * Each script is represented by an json array of block tuples,
+     * where each block tuple is like a line of code. A Sprite may have
+     * multiple Scripts associated with it.  This function returns an array
+     * of json arrays of block tuples for the requested Sprite.
+     * @param spriteName The name of the sprite
+     * @return An array block tuples.  Each block tuple is represented as
+     *         a JSONArray.
+     */
+    private JSONArray[] getScripts(String spriteName) {
+        JSONObject sprite = spriteMap.get(spriteName);
+        if (sprite == null) {
+            throw new IllegalArgumentException(
+                spriteName + " is not the name of a Sprite in this project");
+        }
+        JSONArray scriptTuples = sprite.optJSONArray("scripts");
+        if (scriptTuples == null) {
+            return new JSONArray[0];
```

Comment:
I'm sure there's a good reason to return a zero-length array (rather than null), but I don't know it. @bclinthall, could you enlighten me?